langkit/parsers.py

```diff
@@ -382,6 +398,175 @@ def check_main_rule(self, context):
                     )
                 )
 
+    @classmethod
+    def create_from_lkt(cls, ctx, lkt_unit):
```

Comment:
As discussed live, moving the new compilation logic to a separate file would seem better to me

Reply:
Sure: will do as another step as discussed live.